### ISSUE TITLE
Design and run more complex trajectories in BasicCartesianControl

### ISSUE THREAD

**Issue by jgvictores** [roboticslab-uc3m/kinematics-dynamics#135]
Improve BasicCartesianControl, adding methods to design and run more complex trajectories. For example,
 start with `MoveC`. Blocked by #134. Blocks #133.

Note: Slight possiblity of having to add a new trajectory generation layer, but for now want to try to do within BasicCartesianControl.

**Comment by jgvictores** [comment]
Lots of improvements at f02a0d8283c063eddd347f1ed10ca0221c01ef2d as commented at https://github.com/roboticslab-uc3m/kinematics-dynamics/issues/134#issuecomment-359099379

**Comment by jgvictores** [comment]
Removed `blocked` label, WIP in parallel with #134.

**Comment by PeterBowman** [comment]
Idea: perform a dry run on the desired trajectory (usually `movl`) and check for any limit violations, e.g. maximum joint velocity might be exceeded at some point or the operation would take too long (the latter does not apply to current strategies since trajectory duration is one of our input parameters). On success, perform the trajectory; on failure, return immediately and notify callers.

**Comment by PeterBowman** [comment]
> Idea: perform a dry run on the desired trajectory (usually movl) and check for any limit violations, e.g. maximum joint velocity might be exceeded at some point or the operation would take too long (the latter does not apply to current strategies since trajectory duration is one of our input parameters). On success, perform the trajectory; on failure, return immediately and notify callers.

Moved to #156.

**Comment by jgvictores** [comment]
Not a big fan of doing things for 2 issues anymore. I'm considering this blocked by #134 as it was originally.

**Comment by PeterBowman** [comment]
Given the outcome of https://github.com/roboticslab-uc3m/kinematics-dynamics/issues/134 and https://github.com/roboticslab-uc3m/kinematics-dynamics/issues/168#issuecomment-801017414, we are steering towards the externalization of trajectory generation in BasicCartesianControl. In other words, clients should take care of providing new setpoints (`movi`) to the controller. For instance, one can implement the `MoveC` thing with [`KDL::Path_Circle`](http://docs.ros.org/en/indigo/api/orocos_kdl/html/classKDL_1_1Path__Circle.html), iterate over time and send the usual commands.

@jgvictores are you OK with closing this as WONTFIX, too?

**Comment by jgvictores** [comment]
> @jgvictores are you OK with closing this as WONTFIX, too?

Yep!
